### PR TITLE
Omnissm lambda to autoload elasticsearch on new s3 data

### PR DESCRIPTION
**Pull request by nickerickson**

### CODE REVIEW

File: tools/omnissm/functions/es-upload/main.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			xray.AWS(s3Svc.Client)
+		}
+		if esClient == "" || indexName == "" || typeName == "" {
+			log.Fatal("Missing required env variables OMNISSM_ELASTIC_SEARCH_HTTP, OMNISSM_INDEX_NAME, OMNISSM_TYPE_NAME")
+		}
+		client, err := newElasticClient(esClient)
+		if err != nil {
+			log.Fatal(err)
+		}
+		exists, err := client.IndexExists(indexName).Do(context.Background())
```
Comment (kapilt) [inline tools/omnissm/functions/es-upload/main.go]: doing a round trip network check for a one time initialization in a hot path, seems less then ideal, index setup should just be a makefile target not inline to the event indexing.

Comment (nickerickson) [reply tools/omnissm/functions/es-upload/main.go]: @kapilt I've updated it to have index checking and creation to be done in a separate script.

File: tools/omnissm/functions/es-upload/main.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	keyParams := strings.Split(key, "/")
+	var region string
+	var account string
+	//pull region and/or accountid from key path. there may be a better way to do this
+	for _, element := range keyParams {
+		if strings.Contains(element, "region") {
+			region = strings.Split(element, "=")[1]
+		}
+		if strings.Contains(element, "accountid") {
```
Comment (kapilt) [inline tools/omnissm/functions/es-upload/main.go]: omnissm actually can't use the key path for account id, as we'll launder/aggregate multiple accounts into a single account, as we treat other account resources as hybrid/data center instances. you'll need to parse the inventory data for the custom cloud inventory info to pull this information.

**Comment by nickerickson** [comment]
@kapilt closing as I'm temp pushing some breaking changes. I will remake the PR when its done

**Pull request closed without merging**